src/smc-webapp/project_files.cjsx

```diff
@@ -1525,44 +1525,66 @@ ProjectFilesActionBox = rclass
             if not (single_file_data.is_public and single_file_data.public?.path isnt single_file)
                 @share_click()
 
-    download_click: ->
+    download_single_click: ->
         @props.actions.download_file
             path : @props.checked_files.first()
             log : true
         @props.actions.set_file_action()
 
-    render_download_link: (single_item) ->
+    download_multiple_click: ->
+        destination = ReactDOM.findDOMNode(@refs.download_archive).value
+        @props.actions.zip_files
+            src  : @props.checked_files.toArray()
+            dest : misc.path_to_file(@props.current_path, destination)
+        @props.actions.download_file
```

Comment:
It worked for files I tested with, but they were really small so maybe that's why it worked fine - I will change it just to be safe. 
> the callback is already some default file-listing & cleanup operation
What do you mean by this?